ui/src/main/java/edu/wpi/grip/ui/codeGeneration/Exporter.java

```diff
@@ -0,0 +1,143 @@
+package edu.wpi.grip.ui.codegeneration;
+
+
+import com.google.inject.Singleton;
+
+import org.apache.velocity.Template;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.VelocityEngine;
+import org.apache.velocity.exception.ResourceNotFoundException;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+import java.util.Properties;
+
+import edu.wpi.grip.core.Pipeline;
+import edu.wpi.grip.core.Step;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.OutputSocket;
+import edu.wpi.grip.ui.codegeneration.java.TPipeline;
+
+@Singleton
+public class Exporter {
+  private final String pipelineTemplate = "pipeline.vm";
```

Comment:
It looks like there is a case mismatch between this name and the file "Pipeline.vm". Operating systems other than Windows might not tolerate that.